File: matiasmasso/Mat.NET/DTO/DTOVehicle.cs
using Newtonsoft.Json;

using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOVehicle : DTOGuidNom
    {
        public DTOEmp Emp { get; set; }
        public ModelClass Model { get; set; }
        public string Matricula { get; set; }
        public string Bastidor { get; set; }
        public DTOContact Conductor { get; set; }
        public DTOContact Venedor { get; set; }
        public DateTime Alta { get; set; }
        public DateTime Baixa { get; set; }
        public DTOContract Contract { get; set; }
        public DTOContract Insurance { get; set; }
        public bool Privat { get; set; }
        [JsonIgnore]
        public Byte[] Image { get; set; }

        public DTOVehicle() : base()
        {
        }

        public DTOVehicle(Guid oGuid) : base(oGuid)
        {
        }

        public static DTOVehicle Factory(DTOEmp oEmp)
        {
            DTOVehicle retval = new DTOVehicle();
            {
                var withBlock = retval;
                withBlock.Emp = oEmp;
            }
            return retval;
        }

        public string MarcaYModel()
        {
            System.Text.StringBuilder sb = new System.Text.StringBuilder();
            if (Model != null)
                sb.Append(Model.FullNom() + " ");
            string retval = sb.ToString();
            return retval;
        }

        public string MarcaModelYMatricula()
        {
            string retval = string.Format("{0} {1}", this.MarcaYModel(), Matricula);
            return retval;
        }

        public string MatriculaMarcaYModel()
        {
            string retval = string.Format("{0} {1}", Matricula, this.MarcaYModel());
            return retval;
        }

        public bool IsDonatDeBaixa()
        {
            return Baixa != DateTime.MinValue;
        }


        public class Marca : DTOBaseGuid
        {
            public string Nom { get; set; }
            public List<ModelClass> Models { get; set; }

            public Marca() : base()
            {
                Models = new List<ModelClass>();
            }

            public Marca(Guid oGuid) : base(oGuid)
            {
                Models = new List<ModelClass>();
            }
        }

        public class ModelClass : DTOBaseGuid
        {
            public Marca Marca { get; set; }
            public string Nom { get; set; }

            public ModelClass() : base()
            {
            }

            public ModelClass(Guid oGuid) : base(oGuid)
            {
            }

            public static ModelClass Factory(Marca oMarca)
            {
                ModelClass retval = new ModelClass();
                retval.Marca = oMarca;
                return retval;
            }

            public string FullNom()
            {
                System.Text.StringBuilder sb = new System.Text.StringBuilder();
                if (Marca != null)
                    sb.Append(Marca.Nom + " ");
                sb.Append(Nom);
                string retval = sb.ToString();
                return retval;
            }
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOVehicle.vb
Public Class DTOVehicle
    Inherits DTOGuidNom

    Property emp As DTOEmp
    Property model As DTOVehicleModel
    Property matricula As String
    Property bastidor As String
    Property conductor As DTOContact
    Property venedor As DTOContact
    Property alta As Date
    Property baixa As Date
    Property contract As DTOContract
    Property insurance As DTOContract
    Property privat As Boolean
    <JsonIgnore> Property image As Image

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Shadows Function factory(oEmp As DTOEmp) As DTOVehicle
        Dim retval As New DTOVehicle
        With retval
            .emp = oEmp
        End With
        Return retval
    End Function

    Public Function marcaYModel() As String
        Dim sb As New Text.StringBuilder
        If _model IsNot Nothing Then
            sb.Append(_model.fullNom & " ")
        End If
        Dim retval As String = sb.ToString
        Return retval
    End Function

    Public Function marcaModelYMatricula() As String
        Dim retval As String = String.Format("{0} {1}", Me.marcaYModel, _matricula)
        Return retval
    End Function

    Public Function matriculaMarcaYModel() As String
        Dim retval As String = String.Format("{0} {1}", _matricula, Me.marcaYModel)
        Return retval
    End Function

    Public Function isDonatDeBaixa() As Boolean
        Return _baixa <> Date.MinValue
    End Function
End Class
